cloudinit/sources/DataSourceOVF.py

```diff
@@ -393,20 +403,31 @@ def get_network_config(nics=None, nameservers=None, search=None):
 
 # This will return a dict with some content
 #  meta-data, user-data, some config
-def read_vmware_imc(config):
+def read_vmware_imc(config, paths=None):
     md = {}
     cfg = {}
     ud = None
-    if config.host_name:
-        if config.domain_name:
-            md['local-hostname'] = config.host_name + "." + config.domain_name
-        else:
-            md['local-hostname'] = config.host_name
-
-    if config.timezone:
-        cfg['timezone'] = config.timezone
+    if config is not None:
+        if config.host_name:
+            if config.domain_name:
+                md['local-hostname'] = config.host_name + "." +\
+                                       config.domain_name
+            else:
+                md['local-hostname'] = config.host_name
+        if config.timezone:
+            cfg['timezone'] = config.timezone
+        # Generate a unique instance-id so that new config will be applied
+        md['instance-id'] = "iid-vmware-imc-" + util.rand_str(strlen=8)
```

Comment:
Why is 'instance-id' not provided in the config read from vmware_imc?

Reply:
I think it's same,  the 'instance-id' read from each vmware_imc will be unique to support re-customization.